timesketch/lib/analyzers/login.py

```diff
@@ -0,0 +1,178 @@
+"""Sketch analyzer plugin for login and logoff events."""
+from __future__ import unicode_literals
+
+import logging
+
+from timesketch.lib import emojis
+from timesketch.lib.analyzers import interface
+from timesketch.lib.analyzers import manager
+
+
+LOGON_TYPES = {
+    u'0': u'Unknown',
+    u'2': u'Interactive',
+    u'3': u'Network',
+    u'4': u'Batch',
+    u'5': u'WindowsService',
+    u'7': u'UnlockScreen',
+    u'8': u'NetworkCleartext',
+    u'9': u'NewCredentials',
+    u'10': u'RemoteInteractive',
+    u'11': u'CachedInteractive'
+}
+
+
+def parse_evtx_logoff_event(event, string_list, emoji):
+    """Parse logoff events and return a count of event processed.
+
+    Args:
+        event: an event object (instance of interface.Event).
+        string_list: a list of strings extracted from the Event Log.
+        emoji: string with the Unicode point for the emoji that will be
+            added to the event.
+
+    Returns:
+        int: 1 if the event got processed correctly, 0 otherwise.
+    """
+    emojis_to_add = [emoji]
+    if not len(string_list) == 5:
+        return 0
+
+    attributes = {}
+    attributes['username'] = string_list[1]
+    attributes['logon_domain'] = string_list[2]
+    attributes['session_id'] = string_list[3]
+
+    logon_type_code = string_list[4]
+    attributes['logon_type'] = LOGON_TYPES.get(
+        logon_type_code, LOGON_TYPES.get(u'0'))
+
+    # Want to add an emoji in case this is a screensaver unlock.
+    if logon_type_code == '7':
+        screen_emoji = emojis.get_emoji('screen')
+        emojis_to_add.append(screen_emoji)
+
+    event.add_tags(['logoff-event'])
+    event.add_attributes(attributes)
+    event.add_emojis(emojis_to_add)
+    return 1
+
+
+def parse_evtx_logon_event(event, string_list, string_parsed, emoji):
+    """Parse logon events and return a count of event processed.
+
+    Args:
+        event: an event object (instance of interface.Event).
+        string_list: a list of strings extracted from the Event Log.
+        emoji: string with the Unicode point for the emoji that will be
+            added to the event.
+
+    Returns:
+        int: 1 if the event got processed correctly, 0 otherwise.
+    """
+    attributes = {}
+    emojis_to_add = [emoji]
+
+    if not string_parsed:
+        string_parsed = {}
+        if not len(string_list) == 23:
+            return 0
+        string_parsed['target_user_id'] = string_list[4]
+        string_parsed['target_user_name'] = string_list[5]
+        string_parsed['hostname'] = string_list[11]
```

Comment:
Do we have the src attributes here as well? There is "WorkstationName" and "IpAddress" in 4624 that holds the information where the login came from (localhost or blank for local logins).

Reply:
yes... there are... I'll add them in there as well,... wasn't sure if we should be extracting those as well, but I agree, they can be useful

Reply:
Cool, yeah I use them when I create the graph stuff.

Reply:
I don't see the WorkstationName and/or IpAddress being extracted. Are they present in the 4624 strings?